packages/manager/apps/web/client/app/hosting/hosting.routes.js

```diff
@@ -52,10 +52,41 @@ export default /* @ngInject */ ($stateProvider) => {
           .then((servicesInformation) => servicesInformation.flatten()),
       pendingTasks: /* @ngInject */ (HostingTask, serviceName) =>
         HostingTask.getPending(serviceName).catch(() => []),
+      privateDatabasesIds: /* @ngInject */ (HostingDatabase, serviceName) =>
+        HostingDatabase.getPrivateDatabaseIds(serviceName),
+      privateDatabasesDetachable: /* @ngInject */ (
```

Comment:
It might be a performance issue to add this resolve at the "app.hosting" state. It means every time you go the a child state of app.hosting you will fetch all the services infos of privates databases of the hosting

Do every sub-states of app.hosting requires this information? if no you should move this resolve in the corresponding child state

Reply:
No every time, each time :)

Because if you go from state A to state A.B, it won't be fetched between both, because it has already been fetch, won't it ?

Reply:
Yes if you go from A to A.B it's already fetched but if the data is only required by B then there is no need to load it in A. If everybody is putting data loading at a high level state such as "app.hosting" then the whole application will takes more time to load and it could be easily avoided.

I don't know in the case of this PR if it makes sense but i had the feeling that putting the database fetching queries in the routing of "app.hosting" could be avoided. Hope it makes sense :)

Reply:
I was nitpicking your sentence, your reasoning was correct !

Reply:
@frenautvh, problem is, hosting dashboard requires main routing of hosting, and the actions are in the dashboard so...
But you're right, it may causes performance issues.
To fix that, the general informations dashboard should have its own routing.

I will create a task to do the refactor

Reply:
@frenautvh => MANAGER-4966